chcemvediet/apps/anonymization/management/commands/attachment_anonymization.py

```diff
@@ -0,0 +1,88 @@
+            and stdin is empty, the command will fail. Anonymization created this way will be marked
+            as successful. Only one successful anonymization can be assigned to the Attachment.
+            """)
+
+    option_list = BaseCommand.option_list + (
+        make_option(u'--content_type',
+                    help=squeeze(u"""
+                            Content type of file, e.g. "application/pdf". Automatically guessed from
+                            the file content if not specified.
+                            """)
+                    ),
+        make_option(u'--debug',
+                    default=u'',
+                    help=u'Debug message to the newly created anonymization. Empty by default.'
+                    ),
+        make_option(u'--force',
+                    action=u'store_true',
+                    help=squeeze(u"""
+                            The command refuses to anonymize attachment if a successful
+                            anonymization already exists. This flag disables this check. Deletes all
+                            existing successful anonymizations and creates new one. Unsuccessful
+                            anonymizations will stay unaffected.
+                            """)
+                    ),
+    )
+
+    def handle(self, *args, **options):
+        if not args:
+            raise CommandError(u'attachment_anonymization takes at least 1 argument (0 given).')
+        elif len(args) > 2:
+            raise CommandError(
+                u'attachment_anonymization takes at most 2 arguments ({} given).'.format(len(args))
+            )
+
+        attachment_pk = args[0]
+        try:
+            attachment = Attachment.objects.attached_to(Action).get(pk=attachment_pk)
+        except (Attachment.DoesNotExist, ValueError):
+            raise CommandError(
+                u'Attachment instance with pk "{}" does not exist.'.format(attachment_pk)
+            )
+        attachments_finalization = (AttachmentFinalization.objects
+                                    .filter(attachment=attachment)
+                                    .successful())
+        if not options[u'force'] and attachments_finalization:
+            raise CommandError(squeeze(u"""
+                    Anonymization files already exist. Use the --force option to overwrite
+                    them.
+                    """))
+
+        if len(args) == 2:
+            filename = args[1]
+            try:
+                with open(filename, u'rb') as file:
+                    content = file.read()
+            except IOError as e:
+                raise CommandError(u'Could not open file: {}.'.format(e))
+        else:
+            content = sys.stdin.read()
+            if not content:
+                raise CommandError(u'No content given.')
+
+        attachments_finalization.delete()
+        AttachmentFinalization.objects.create(
```

Comment:
@viliambalaz Este by sa to cele zislo zabalit do tranzakcie. Pre pripad, ze ak nieco padne po `delete`, tak nech sa cela tranzakcia rollbackne. Aby sa nestalo, ze stary finalization sa zmaze a novy sa koli chybe nevytvori.

Vid napriklad `apps/inforequests/cron.py` ako sa pouziva `with transaction.atomic():` pre tranzakcie.

Reply:
:ok: @martinmacko47 Len pre vysvetlenie, z akeho dovodu sa  v `apps/inforequests/cron.py` pouziva aj dekorator aj context manager? Kedze exception sami zachytavame a funkcia skonci bez chyby, musime pouzit context_manager. 

Preco je potom pouzity aj dekorator? Stav databazy sa meni iba vo vnutri context managera.

Reply:
@viliambalaz To je dobra otazka. Zvycajne staci jeden atomic. Vnoreny atomic su vnorene tranzakcie.

V `undecided_email_reminder` v `apps/inforequests/cron.py` bolo treba vnorene tranzakcie, aby ked failne jedno volanie `send_undecided_email_reminder`, tak aby sa upratalo vsetko co sa pocas toho volania `send_undecided_email_reminder` vytvorilo. Ale zaroven aby sa nezmazalo nic, co sa vytvorilo pocas ostatnych volani `send_undecided_email_reminder` v tom istom cykle. Vonkajsia tranzakcia tam sluzi na to, aby sa zarucila konzistencia precitanych dat z DB. Teda aby medzi selectnutim a prefiltrovanim `inforequests` a ich spracovanim v `send_undecided_email_reminder` v DB nepribudli ziadne nove data. Pretoze ak by pribudli, tak by sme mohli poslat nejaku blbost v maile.

V nasom pripade nam vnorene tranzakcie netreba, kedze robime iba jednu vec. Chceme aby cely command bol bud uspesny, alebo ak failne, tak aby sa po nom vsetko upratalo. Cize staci vonkasia `@transaction.atomic`

Reply:
:ok: